### PR TITLE
Bug Fix: mbedtls_ecdsa_verify_restartable fails with ECDSA_SIGN_ALT

### LINKED ISSUE #7498: mbedtls_ecdsa_verify_restartable fails with ECDSA_SIGN_ALT
When ECDSA_SIGN_ALT but not ECDSA_VERIFY_ALT, mbedtls_ecdsa_can_do was not being defined causing mbedtls_ecdsa_verify_restartable to always fail

### PR DESCRIPTION
**Pull request by JonathanWitthoeft**
When ECDSA_SIGN_ALT but not ECDSA_VERIFY_ALT, mbedtls_ecdsa_can_do was not being defined causing mbedtls_ecdsa_verify_restartable to always fail

closes #7498

## Description

This will support ATECC608 integration for signing but not verification.

## Gatekeeper checklist

- [x] **changelog** provided, or not required
- [ ] **backport** done, or not required
- [ ] **tests** provided, or not required

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

### CODE REVIEW

**Comment by gilles-peskine-arm** [comment]
Out of curiosity, which ATECC608 integration are you referring to? I don't recall seeing one that uses `ECDSA_SIGN_ALT`.

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
I don't think the change is quite right. Also it needs a changelog entry.

File: library/ecdsa.c [hunk 1]
```diff
@@ -234,6 +234,24 @@ static int derive_mpi(const mbedtls_ecp_group *grp, mbedtls_mpi *x,
 }
 #endif /* ECDSA_DETERMINISTIC || !ECDSA_SIGN_ALT || !ECDSA_VERIFY_ALT */
 
+#if !defined(MBEDTLS_ECDSA_SIGN_ALT)     || \
```
[the remark this answers is not in the document]
Comment (JonathanWitthoeft) [reply library/ecdsa.c]: I removed the conditions

File: library/ecdsa.c [hunk 2]
```diff
@@ -234,6 +234,24 @@ static int derive_mpi(const mbedtls_ecp_group *grp, mbedtls_mpi *x,
 }
 #endif /* ECDSA_DETERMINISTIC || !ECDSA_SIGN_ALT || !ECDSA_VERIFY_ALT */
```
[the remark this answers is not in the document]
Comment (JonathanWitthoeft) [reply library/ecdsa.c:236]: I pushed ChangeLog.d/mbedtls_ecdsa_can_do-unconditional-define.txt

File: ChangeLog.d/mbedtls_ecdsa_can_do-unconditional-define.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Bugfix
+   * Removes !ECDSA_SIGN_ALT condition around mbedtls_ecdsa_can_do 
+     definition, so that mbedtls_ecdsa_verify_restartable will not 
+     automatically fail.
```
Comment (gilles-peskine-arm) [inline ChangeLog.d/mbedtls_ecdsa_can_do-unconditional-define.txt:4]: This describes the change, but changelog entries are primarily meant for users, so they should explain why the change is needed. Also, please don't abbreviate symbols here: it's not necessarily obvious to readers how to complete them. Also the entry should mention the issue number when there is one. So something like

> Fix a build error when MBEDTLS_ECDSA_SIGN_ALT is defined but not MBEDTLS_ECDSA_VERIFY_ALT. Fixes #7498.

Comment (JonathanWitthoeft) [reply ChangeLog.d/mbedtls_ecdsa_can_do-unconditional-define.txt]: Is the updated ChangeLog better?

**Review by gilles-peskine-arm — APPROVED** [review]
Thanks for updating, looks good to me!

**Review by tom-daubney-arm — APPROVED** [review]
LGTM.

Thanks @JonathanWitthoeft !
